web/src/components/checkbox/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useRef, useState } from 'react'
+import { useTheme } from 'react-jss'
+import clsx from 'clsx'
+import { VisuallyHidden } from '@react-aria/visually-hidden'
+import { useFocusRing } from '@react-aria/focus'
+import { useToggleState } from '@react-stately/toggle'
+import { useCheckbox } from '@react-aria/checkbox'
+
+import { genUid } from '../../utils'
+import { useStyles } from './styles'
+import { useFocusStyle } from '../../styles/global'
+import { Text } from '../typography'
+import { CheckmarkIcon, IndeterminateIcon } from './icons'
+
+export interface Props {
+  id?: string
+  checked?: boolean
+  invalid?: boolean
+  disabled?: boolean
+  required?: boolean
+  indeterminate?: boolean
+}
+
+export interface WithLabel extends Props {
+  label: string
+  helpText?: string
+}
+export interface CheckboxOnly extends Props {
+  ariaLabel: string
+}
+
+export type CheckboxProps = CheckboxOnly | WithLabel
+
+const isLabelCheckbox = (variableToCheck): variableToCheck is WithLabel =>
+  (variableToCheck as WithLabel).label !== undefined
+
+export const Checkbox: React.FC<CheckboxProps> = props => {
+  const { checked, disabled, indeterminate, invalid, id } = props
+  const state = useToggleState({
+    ...props,
+    isSelected: checked,
+  })
+  const ref = useRef()
+  const [_id] = useState(id || genUid())
+  const { inputProps } = useCheckbox(
+    { ...props, isDisabled: disabled, isIndeterminate: indeterminate },
+    state,
+    ref,
+  )
+  const { isFocusVisible, focusProps } = useFocusRing()
+  const theme = useTheme()
+  const focusClass = useFocusStyle(props)
+  const classes = useStyles({
+    ...props,
+    theme,
+    isFocusVisible,
+    checked: state.isSelected,
+  })
+  const hasLabel = isLabelCheckbox(props)
```

Comment:
This is because of typescript's type guards.

We have 2 different prop groups

"WithLabel" and "CheckboxOnly"

so you can use the component in two different ways.

If you provide a label to the component, you can provide a helptext aswell.

If arialabel is provided, not.